lib/dry/validation/messages/resolver.rb

```diff
@@ -69,6 +69,12 @@ def message(rule, tokens: EMPTY_HASH, locale: nil, full: false, path:)
             template, meta = messages[rule, msg_opts.merge(path: keys.last)] unless template
           end
 
+          if !template && keys.size > 1
+            non_index_keys = []
+            keys.each { |k| non_index_keys << k unless k.is_a?(Numeric) }
```

Comment:
these two lines are `non_index_keys = keys.reject { |k| k.is_a?(Integer) }`

Reply:
lol, I tried to use `filter` but it was unavailable in the 2.4 version and I totally forgot about `reject`, thanks :)

Reply:
ruby's version of `filter` is `select` which IMO makes more sense because it's not obvious if `filter` filters _in_ or _out_